Source/WebKit/UIProcess/ProcessThrottler.h

```diff
@@ -143,6 +143,14 @@ class ProcessThrottler : public CanMakeWeakPtr<ProcessThrottler> {
     void setAllowsActivities(bool);
     void setShouldTakeSuspendedAssertion(bool);
     void delaySuspension();
+    bool isSuspended() const
+    {
+#if PLATFORM(COCOA)
```

Comment:
I wish we wouldn't add build checks in this cross-platform code. I think we should make sure that other ports construct ProcessAssertions when needed. ProcessAssertion is cross-platform (though is a no-op on non-Cocoa ports at the moment).

I suggest we double check that GTK for example construct ProcessAssertions and ping a GTK port engineer for extra validation (E.g. @mcatanzaro).

Reply:
We only drop all assertions on Cocoa so `isSuspended()` should always return false on other platforms. I double checked that GTK is able to construct ProcessAssertions. I agree extra validation would be great. I'll ping @mcatanzaro